### PR TITLE
safePosition - aax

### PR DESCRIPTION
**Pull request by ttodua**
the calculations (of removed fields ) are done in base safePosition

### CODE REVIEW

File: js/base/Exchange.js [hunk 1]
```diff
@@ -1682,6 +1682,195 @@ module.exports = class Exchange {
...
+        const lastPriceString = this.safeString (position, 'lastPrice');
+        // support deprecated 'marginType'
+        const marginMode = this.safeString2 (position, 'marginMode', 'marginType');
+        //
+        // CONTRACT SIZE
+        //
+        let contractSizeString = this.safeString (position, 'contractSize');
+        if (contractSizeString === undefined) {
+            contractSizeString = this.safeString (market, 'contractSize');
+        }
```
Comment (kroitor) [inline js/base/Exchange.js:1700]:
```suggestion
        const marketContractSize = this.safeString (market, 'contractSize')
        const contractSizeString = this.safeString (position, 'contractSize', marketContractSize);
```

Comment (ttodua) [reply js/base/Exchange.js:1700]: for the case of simplicity of the PR/edits, I prefer not to touch other things, even simple edits. let's leave them as they are, and after merging I'll re-do the optimization of the codes.

Comment (ttodua) [reply js/base/Exchange.js:1700]: one slight note - the above suggestion ( i missed it) was related to #12765 , so you can comment there.
second slight note - i think there is tiniest performance gain if we don't access/parse `market` object, if the contract-size was already defined, thus ` === undefined` here is better imho.

**Comment by frosty00** [comment]
#15856

**Pull request closed without merging**